README.md

```diff
@@ -143,69 +264,114 @@ Implementing push notifications requires a few additional snippets of code to en
+From here on depending on which dependency manager you use the steps would look slightly different:
 
-The following code example allows you to track when a user opens a push notification.
+<details>
+<summary> Swift Package Manager(SPM) </summary>
 
-1. Add the following code that extends your app delegate:
+- Tap on the newly created notification service extension target
+  - Under General > Frameworks and libraries add `KlaviyoSwiftExtension` using the + button at the bottom left.
+  - Then in the `NotificationService.swift` file add the code for the two required delegates from [this](Examples/KlaviyoSwiftExamples/SPMExample/NotificationServiceExtension/NotificationService.swift) file. This sample covers calling into Klaviyo so that we can download and attach the media to the push notification.
+</details>
 
-```swift
-extension AppDelegate: UNUserNotificationCenterDelegate {
-    func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) {
-	let handled = KlaviyoSDK().handle(notificationResponse: response, withCompletionHandler: completionHandler)
-	if !handled {
-	   // not a klaviyo notification should be handled by other app code
-	}
-    }
-}
+<details>
+<summary> Cocoapods </summary>
 
-```
+- In your `Podfile` add in `KlaviyoSwiftExtension` as a dependency to the newly added notification service extension target.
 
-Once your first push notifications are sent and opened, you should start to see _Opened Push_ metrics within your Klaviyo dashboard.
+  Example:
 
-### Foreground push handling
+  ```
+  target 'NotificationServiceExtension' do
+      pod 'KlaviyoSwiftExtension', '2.1.0-beta1'
+  end
+  ```
 
-The following code example allows push notifications to be displayed when your app is running:
+  Be sure to replace the name of your notification service extension target above.
 
-```swift
+  - Once you've added in the dependency make sure to `pod install`.
+  - Then in the `NotificationService.swift` file add the code for the two required delegates from [this](Examples/KlaviyoSwiftExamples/CocoapodsExample/NotificationServiceExtension/NotificationService.swift) file. This sample covers calling into Klaviyo so that we can download and attach the media to the push notification.
+</details>
 
-    func userNotificationCenter(_ center: UNUserNotificationCenter,
-                                  willPresent notification: UNNotification,
-                                  withCompletionHandler completionHandler: @escaping (UNNotificationPresentationOptions) -> Void) {
-        var options: UNNotificationPresentationOptions =  [.alert]
-        if #available(iOS 14.0, *) {
-          options = [.list, .banner]
-        }
-        completionHandler(options)
-    }
-```
+* Test your rich push notifications
```

Comment:
It feels like we're missing a sentence or two after this. Are the next lines bullets for how someone could test? Should we say something like "To test rich push notifications, you will need three things:"

Reply:
added